src/devtools/client/webconsole/selectors/messages.js

```diff
@@ -31,12 +33,22 @@ export const getVisibleMessages = createSelector(
   state => state.messages.visibleMessages,
   state => state.consoleUI.zoomStartTime,
   state => state.consoleUI.zoomEndTime,
-  (messages, visibleMessages, zoomStartTime, zoomEndTime) => {
-    return visibleMessages.filter(id => {
+  getTrimRegion,
+  (messages, visibleMessages, zoomStartTime, zoomEndTime, trimRegion) => {
```

Comment:
Will this run on every update? If so, it's probably better to use `useMemo` in the components that care about filtering.

Reply:
Yep, I missed that this was happening inside of `createSelector` because of the way the diff was cut off. All good to go 💯